Grammar improvements

rewinfrey opened the pull request:
This patch adds improvements to the JavaScript grammar while working on https://github.com/tree-sitter/tree-sitter-typescript/pull/28. 

/cc @github/semantic-code @maxbrunsfeld

maxbrunsfeld commented on grammar.js:
```diff
@@ -775,7 +806,7 @@ module.exports = grammar({
     _property_name: $ => choice($.identifier, $.reserved_identifier, $.string, $.number),
 
-    reserved_identifier: $ => choice('get', 'set', 'async')
+    reserved_identifier: $ => choice('get', 'set', 'async', 'abstract', 'module', 'interface'),
```
Should these just be added in the typescript grammar? I'd think it'd be ok to just let these words be parsed as regular `identifiers` in regular JavaScript.

rewinfrey replied:
@maxbrunsfeld I was basing this off this list: https://www.w3schools.com/js/js_reserved.asp. I just realized though that `module` is not a JavaScript keyword. I'll move `module` over to the TypeScript grammar, but I think `abstract` and `interface` are good here. What do you think?

rewinfrey commented:
@maxbrunsfeld thanks for the comment. I updated the grammar to remove `module` from the `reserved_identifier` list, and added `module` to the TypeScript grammar's `reserved_identifier` list. 🙇
